addons/mrp/models/product.py

```diff
@@ -232,8 +232,12 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
         # pre-compute bom lines and identify missing kit components to prefetch
         bom_sub_lines_per_kit = {}
         prefetch_component_ids = set()
-        for product in bom_kits:
-            __, bom_sub_lines = bom_kits[product].explode(product, 1)
+        # memoize explode calls as we are using bom_sub_lines that depends only on the bom.
+        exploded_values = {}
+        for product, bom in bom_kits.items():
+            if not exploded_values.get(bom):
+                exploded_values[bom] = bom_kits[product].explode(product, 1)
+            __, bom_sub_lines = exploded_values[bom]
```

Comment:
I'm really not sure about this one :confused: 
Consider the following Kit: 
<img width="1222" height="557" alt="image" src="https://github.com/user-attachments/assets/bc830a08-22a0-45de-95b7-45eab5cd51cb" />

Indeed you'll have thrice the same bom, but each explode would give different `bom_sub_lines`, as some lines are variant-specific.